packages/mds-audit/api.ts

```diff
@@ -188,12 +191,16 @@ function api(app: express.Express): express.Express {
             })
 
             // Create the audit start event
+            const provider_event = await readEvent(provider_device_id)
             await writeAuditEvent({
               audit_trip_id,
               audit_event_id,
               audit_subject_id,
               audit_event_type: AUDIT_EVENT_TYPES.start,
               ...flattenTelemetry(telemetry),
+              provider_event_id: provider_event ? provider_event.id : null,
+              provider_event_type: provider_event ? provider_event.event_type : null,
+              provider_event_type_reason: provider_event ? provider_event.event_type_reason : null,
```

Comment:
This sequence of lines appear a number of times. Might be a candidate for a `...flattenProviderEvent(provider_event)` treatment similar to telemetry.

Reply:
Good call